src/tests/unit/components/app-content-switcher/AppContentSwitcher.spec.jsx

```diff
@@ -0,0 +1,45 @@
+import { render, screen, fireEvent } from '@testing-library/react'
+import AppContentSwitcher from '~/components/app-content-switcher/AppContentSwitcher'
+import { vi } from 'vitest'
+
+const defaultProps = {
+  active: true,
+  onChange: vi.fn(),
+  switchOptions: {
+    left: { text: 'Left Option', tooltip: 'Left Tooltip' },
+    right: { text: 'Right Option', tooltip: 'Right Tooltip' }
+  }
+}
+
+describe('AppContentSwitcher', () => {
+  it('should render with the correct props', () => {
+    render(<AppContentSwitcher {...defaultProps} />)
+
+    expect(screen.getByText('Left Option')).toBeInTheDocument()
+    expect(screen.getByText('Right Option')).toBeInTheDocument()
+    expect(screen.getByTestId('switch')).toBeInTheDocument()
+  })
+
+  it('should call the onChange function when the switch is clicked', () => {
+    render(<AppContentSwitcher {...defaultProps} />)
+
+    const switchElement = screen.getByTestId('switch').querySelector('input')
+    fireEvent.click(switchElement)
```

Comment:
we should prefer userEvent vs fireEvent where it is possible
https://testing-library.com/docs/user-event/intro/#difference-to-fireevent
![image](https://github.com/user-attachments/assets/0d92616a-7ac8-4932-931c-af2ff1214441)